core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -606,6 +608,9 @@ public synchronized void updateMetaData(final IndexMetaData metadata) {
             if (refreshTask.getInterval().equals(indexSettings.getRefreshInterval()) == false) {
                 rescheduleRefreshTasks();
             }
+            if (trimTranslogTask.getInterval().equals(indexSettings.getTranslogRetentionCheckInterval()) == false) {
```

Comment:
> Can you elaborate why you go down this route instead of using IndexShard#afterWriteOperation() 

This is not instead but on top of it. The problem is that if people have a 12h retention policy and  we need to start cleaning stuff 12hs later (assuming they index full speed and roll over to a new index).

>  I'd love to rather remove tasks than adding them?! 

I'm totally with you. I spent some time thinking about ways to avoid the regular check but everything I came up with ended up more complex. I considered stuff scheduling things in in activity etc.. always come back to this being the clearest/easiest. I'll be more than happy to hear about alternatives.